.github/workflows/build.yml

```diff
@@ -10,30 +10,36 @@ on:
 jobs:
   build:
     env:
-        NUGET_API_KEY: ${{ secrets.NUGET_API_KEY }}
-        
-    runs-on: ${{ matrix.os }}
+      NUGET_API_KEY: ${{ secrets.NUGET_API_KEY }}
+    runs-on: ubuntu-latest
+    container: debian:oldoldstable
     strategy:
       fail-fast: true
-      matrix: 
-        os: [ubuntu-latest]
 
     steps:
-    - name: Install X11 dependencies
-      run: |
-        sudo apt-get update
-        sudo apt-get install xorg-dev libwayland-dev libxkbcommon-dev wayland-protocols extra-cmake-modules
-    - uses: actions/checkout@v3
-    - name: Setup .NET Core
-      uses: actions/setup-dotnet@v3
-      with:
-        dotnet-version: 6.0.x
-    - name: Download dependencies and create Nuget Package
-      run: ./make_nuget.ps1 "" ${{ github.run_number }}
-      shell: pwsh
-    - uses: actions/upload-artifact@v3
-      with:
-        name: NugetPackage
-        path: ./artifacts/*.nupkg
-    - name: Deploy Nuget Package
-      run: dotnet nuget push ./artifacts/*.nupkg -k $NUGET_API_KEY -s https://api.nuget.org/v3/index.json
+      - name: Install X11 dependencies
+        run: |
+          apt-get update
+          apt-get install -y xorg-dev libwayland-dev libxkbcommon-dev wayland-protocols extra-cmake-modules
+      - name: Install build dependencies
+        run: |
+          apt-get install -y curl wget git cmake
+          VERSION_ID=$(awk -F= '/^VERSION_ID=/{gsub(/"/, "", $2); print $2}' /etc/os-release)
+          wget -q https://packages.microsoft.com/config/debian/$VERSION_ID/packages-microsoft-prod.deb
+          dpkg -i packages-microsoft-prod.deb
+          apt-get update
+          apt-get install -y powershell
```

Comment:
Why is it suddenly necessary to install powershell manually? Does it not come with the container?

Reply:
GitHub actions are a bit weird, basically GitHub installs all one could need into the ubuntu image they provide

when using a container it runs that container next to the ubuntu image and then the commands inside the container which is just the bare bones debian docker container from https://hub.docker.com/_/debian (~50MB compressed)
so we need to install everything we need in there

https://docs.github.com/en/actions/using-jobs/running-jobs-in-a-container